libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton.tsx

```diff
@@ -0,0 +1,396 @@
+     *
+     * Note: this is optional and deprecated. If you do not specify this, then you MUST specify the filter prop.
+     *
+     * @deprecated - use the filter prop instead
+     */
+    identifier?: string;
+
+    /**
+     * Optionally specify title for the attribute filter. By default, the attribute name will be used.
+     */
+    title?: string;
+
+    /**
+     * Specify function which will be called when user clicks 'Apply' button. The function will receive the current
+     * specification of the filter, as it was updated by the user.
+     *
+     * @param filter - new value of the filter.
+     */
+    onApply: (filter: IAttributeFilter, isInverted: boolean) => void;
+
+    /**
+     * Optionally customize attribute filter with a callback function to trigger when an error occurs while
+     * loading attribute elements.
+     */
+    onError?: (error: any) => void;
+}
+
+/**
+ * @public
+ */
+export type IAttributeFilterButtonProps = IAttributeFilterButtonOwnProps & WrappedComponentProps;
+
+const DefaultFilterError: React.FC = injectIntl(({ intl }) => {
+    const text = intl.formatMessage({ id: "gs.filter.error" });
+    return <div className="gd-message error s-button-error">{text}</div>;
+});
+
+const DropdownButton: React.FC<{
+    isMobile?: boolean;
+    isOpen?: boolean;
+    title: string;
+    subtitle: string;
+}> = ({ isMobile, isOpen, title, subtitle }) => {
+    return (
+        <div
+            className={cx("gd-attribute-filter-button", {
+                "is-active": isOpen,
+                "gd-attribute-filter-button-mobile": isMobile,
+            })}
+        >
+            <div className="button-content">
+                <div className="button-title">{title}</div>
+                <div className="button-subtitle">{subtitle}</div>
+            </div>
+        </div>
+    );
+};
+
+const LIMIT = MAX_SELECTION_SIZE + 50;
+
+export const AttributeFilterButtonCore: React.FC<IAttributeFilterButtonProps> = (props) => {
+    const [validElements, setValidElements] = useState<IElementQueryResultWithEmptyItems>(null);
+    const [selectedFilterOptions, setSelectedFilterOptions] = useState<Array<IAttributeElement>>([]);
+    const [isInverted, setIsInverted] = useState(true);
+    const [prevSelectedFilterOptions, setPrevSelectedFilterOptions] = useState<Array<IAttributeElement>>([]);
+    const [prevIsInverted, setPrevIsInverted] = useState(true);
+    const [firstLoad, setFirstLoad] = useState(true);
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState<any>(null);
+    const [searchString, setSearchString] = useState("");
+    const [offset, setOffset] = useState(0);
+    const [limit, setLimit] = useState(LIMIT);
+    const [totalCount, setTotalCount] = useState(LIMIT);
+    const [filterOptions, setFilterOptions] = useState<Array<IAttributeElement>>([]);
+    const [title, setTitle] = useState("");
+    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
```

Comment:
Maybe only 1 useState with state documented by some interface for everything would be easier to think about?